### ISSUE TITLE
Navigating away from a focused CodeMirror blocks all inputs on the page

### ISSUE THREAD

**Issue by Hypnosphi** [uiwjs/react-codemirror#677]
Example on CodeSandbox: https://codesandbox.io/p/devbox/react-codemirror-breaks-input-5sl4fw

To reproduce, click on "CodeMirror" link in the preview and then navigate back using the preview browser controls (that is, without blurring the CodeMirror element). After that, no input in the textarea is possible

**Comment by jaywcjlove** [comment]
https://github.com/user-attachments/assets/acd30e2f-7217-4099-ad64-b6bd8bbc177b

@Hypnosphi  After testing, I can edit. I’m not sure if I misunderstood.

**Comment by Hypnosphi** [comment]
@jaywcjlove it's because the CodeMirror gets blurred when you click "TextArea" link. To reproduce, please use the browser "back" button above

**Comment by jaywcjlove** [comment]
I reproduced the error. I think if you need to put the `textarea` in a component and maintain a state, it might solve the problem.

**Comment by Hypnosphi** [comment]
@jaywcjlove No, it blocks absolutely all inputs, controlled or uncontrolled, and even the ones not rendered by React:

https://codesandbox.io/p/devbox/react-codemirror-breaks-input-5sl4fw

**Comment by helmutmigge** [comment]
Also noticed this happening here

**Comment by Hypnosphi** [comment]
Let me know if you find a workaround like performing some cleanup on unmounting codemirror

**Comment by jazeee** [comment]
I've seen this too. It seems to be dependent on how the component is managed in React code.
It locks every input field. The only workaround I have seen is that you can reopen the view that contains the ReactCodeMirror component, focus on it, and then blur away, and then go back to what you were on.
I am trying to reproduce it in a simple way. (I observe it without routing involved, but rather with conditionally showing the a side drawer that contains ReactCodeMirror.

**Comment by jazeee** [comment]
More specific observation
1. Open a drawer with a container contianing an editable, focusable `ReactCodeMirror`. (It must get focus)
2. Swap out the container contents to a different form. The inputs in that other form don't allow you to type. Nor do any other inputs in the dom. You can hit the back arrow to see the cursor move, but you cannot type anything.

https://github.com/user-attachments/assets/70e9309a-fa14-4298-aaf9-0c103a57073d

**Comment by jazeee** [comment]
We have seem intermittent issues without portals or drawers, but in order to reliably see the issue, here's a reproducible version:
https://codesandbox.io/p/live/edbd7977-c460-48db-b3ea-c44807071ffe
Generally, there appears to be issues related to grabbing input field focuses, and releasing focus.

To see the issue,
1. focus on the code mirror within a drawer
2. Click on a button outside of the drawer that sets focus on another input
3. Attempt to type in any input field.
Observe that the code mirror field has focus and gets data entry
Expect that the focused input field gets data entry.

**Comment by jazeee** [comment]
Looking at the react-codemirror code, I see a prop `root`, related to portals. That doesn't seem to help.
In addition, there are `useEffect` functions that seem to introduce repeated calls to codemirror.
One thing that may help things:
```
      if (!view) {
        const viewCurrent = new EditorView({
          state: stateCurrent,
          parent: container,
          root,
        });
        setView(viewCurrent);
//...
     }
```
should be something like:
```
setView((currentView) => {
  if (currentView) {
    return currentView
  }
  return new EditorView... 
}
```
I don't know if this will solve the issue, but generally, updating state based on current state should be done functionally since the latest state is not updated immediately.

**Comment by jazeee** [comment]
Video example of 
https://codesandbox.io/p/live/edbd7977-c460-48db-b3ea-c44807071ffe

https://github.com/user-attachments/assets/64171edb-185e-45b0-a963-de0be6ccb033

**Comment by jazeee** [comment]
I fixed the codesandbox link:
https://codesandbox.io/p/devbox/react-codemirror-breaks-input-forked-l7rhtv

**Comment by jaywcjlove** [comment]
> Looking at the react-codemirror code, I see a prop `root`, related to portals. That doesn't seem to help. In addition, there are `useEffect` functions that seem to introduce repeated calls to codemirror. One thing that may help things:
> 
> ```
>       if (!view) {
>         const viewCurrent = new EditorView({
>           state: stateCurrent,
>           parent: container,
>           root,
>         });
>         setView(viewCurrent);
> //...
>      }
> ```
> 
> should be something like:
> 
> ```
> setView((currentView) => {
>   if (currentView) {
>     return currentView
>   }
>   return new EditorView... 
> }
> ```
> 
> I don't know if this will solve the issue, but generally, updating state based on current state should be done functionally since the latest state is not updated immediately.

@jazeee  Contributions via PR to solve this issue are warmly welcomed!

**Comment by jaywcjlove** [comment]
@jazeee I couldn’t reproduce this error in Next.js, and I still haven’t figured out the exact cause of the issue.

**Comment by jazeee** [comment]
@jaywcjlove yeah, it is a tricky bug. Here is the code sample I was able to repro in.
[Codesandbox](https://codesandbox.io/p/devbox/react-codemirror-breaks-input-forked-l7rhtv)
[Video](https://github.com/uiwjs/react-codemirror/issues/677#issuecomment-2427916848)
The bug seems to be related to how the component (or the underlying codemirror code) is rendered.
In my repro, it is inside a MUI Drawer, but others appear to have seen this bug during client side react-router-dom route changes.
(May need client side rendering to help see it more often)

**Comment by vezaynk** [comment]
Duplicate of https://github.com/codemirror/dev/issues/1458
